python/pyarrow/table.pxi

```diff
@@ -671,13 +671,61 @@ cdef class RecordBatch(_PandasConvertible):
         Returns
         -------
         RecordBatch
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> pydict = {'int': [1, 2], 'str': ['a', 'b']}
+        >>> pa.RecordBatch.from_pydict(pydict)
+        pyarrow.RecordBatch
+        int: int64
+        str: string
         """
 
         return _from_pydict(cls=RecordBatch,
                             mapping=mapping,
                             schema=schema,
                             metadata=metadata)
 
+    @staticmethod
+    def from_pylist(mapping, schema=None, metadata=None):
+        """
+        Construct a RecordBatch from list of dictionary of rows.
```

Comment:
```suggestion
        Construct a RecordBatch from list of rows / dictionaries.
```

Each dictionary represents a row, so "dictionary of rows" sounds a bit strange ("dictionary of row values" could be strictly speaking more correct, but I still find that not super clear)

Reply:
Totally agree. +1 for the suggestion.